internal/install/recipe_installer.go

```diff
@@ -138,6 +144,17 @@ func (i *RecipeInstaller) discoverAndRun(ctx context.Context) error {
 	return i.guidedInstall(ctx, m)
 }
 
+func (i *RecipeInstaller) assertDiscoveryValid(ctx context.Context, m *types.DiscoveryManifest) error {
+	if m.OS == "" {
+		return fmt.Errorf("failed to identify a valid operating system")
+	}
+	if !(strings.ToLower(m.OS) == "linux" || strings.ToLower(m.OS) == "windows") {
+		return fmt.Errorf("operating system %s is not supported", m.OS)
+	}
+	log.Debugf("Done asserting valid operating system for %s", m.OS)
+	return nil
+}
```

Comment:
I think we still need a check in here for when `platform` is empty

Reply:
@jbeveland27 I'm not sure about an empty check on platform. I could see recipes just defining OS without platform.
We can always add more rules later when we're sure. For now, I'd like to get this enhancement out to protect the cases we know are not supported (darwin).